bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/MapUtils.java

```diff
@@ -148,9 +151,59 @@ public static void checkIsMapOnlyOperation(BType mapType, String op) {
         switch (mapType.getTag()) {
             case TypeTags.MAP_TAG:
             case TypeTags.JSON_TAG:
+            case TypeTags.RECORD_TYPE_TAG:
                 return;
             default:
                 throw createOpNotSupportedError(mapType, op);
         }
     }
+
+    public static void checkValidFieldForRecord(MapValue<?, ?> m, String  k, String  op) {
+        BType type = m.getType();
+        switch (type.getTag()) {
+            case TypeTags.RECORD_TYPE_TAG:
+                boolean isValid = checkField(m, k);
+                if (!isValid) {
+                    return;
+                }
+                boolean isRequired = checkForRequiredFields((BRecordType) type, k);
+                if (isRequired) {
+                    throw createOpNotSupportedError(type, op);
```

Comment:
Should we add more info to say it failed since it is a required field?

Reply:
+1 will add